python_modules/libraries/dagster-dg/dagster_dg/scaffold.py

```diff
@@ -152,6 +113,73 @@ def scaffold_project(
             RemoteComponentRegistry.from_dg_context(cl_dg_context)  # Populate the cache
 
 
+# Despite the fact that editable dependencies are resolved through tool.uv.sources, we need to set
+# the dependencies themselves differently depending on whether we are using editable dagster or
+# not. This is because `tool.uv.sources` only seems to apply to direct dependencies of the package,
+# so any 2+-order Dagster dependency of our package needs to be listed as a direct dependency in the
+# editable case.
```

Comment:
Link to the relevant github issue in the uv repo in the comment

Reply:
https://github.com/astral-sh/uv/issues/9446 because i had it handy